### DISCUSSION TITLE
VPS Perf

### DISCUSSION THREAD

**Discussion by rolanday** [vikejs/vike#1093] [Discussion] [answered]
Ran `hey -n 100000 http://localhost:3000` on VPS and take about 60s to complete compared to 25s for Remix -- both using Express and for route just returning "hello". I realize not a real-world benchmark, but easy to setup and a datapoint nonetheless.  Being new to VPS, my instinct was VPS should win on perf since compared to frameworks like Remix/Nextjs, but doesn't seem to be the case. What are others seeing?

Thanks!

**Comment by brillout** [comment, marked as answer]
This isn't a reliable way to measure DX performance. AFAIK Remix builds everything before spinning up a dev server, so obviously things will be faster after Remix's initial slow startup which ends up being *very* slow on large scale apps.

See https://vite-plugin-ssr.com/lazy-transpiling.

**Comment by rolanday** [comment]
Agreed not a reliable measure of DX, but nonetheless found the 2x diff to be unexpected because both are running on Express (w/ minimal middleware) so curious if any insights as to why. I ran the test on production builds, so no dev server involved, but you are correct that remix builds everything before spinning up dev server. Thanks for this great contrib btw!! It's a very exciting project and great to have a Vite opting for SSR'ing React project.
